src/ELF/Builder.tcc

```diff
@@ -827,6 +827,12 @@ void Builder::build_symbol_hash(void) {
   hashtable_stream.setpos(0);
   uint32_t nbucket = hashtable_stream.read_conv<uint32_t>();
   uint32_t nchain  = hashtable_stream.read_conv<uint32_t>();
+  if (nchain != this->binary_->dynamic_symbols_.size()) {
```

Comment:
I'm curious about the reason.  
Do you have a reference of such requirement ?

Reply:
According to [Oracle: Hash Table Section](https://docs.oracle.com/cd/E23824_01/html/819-0690/chapter6-48031.html):

> The bucket array contains nbucket entries, and the chain array contains nchain entries. Indexes start at 0.
> The number of symbol table entries should equal nchain, so symbol table indexes also select chain table entries.

And according to [flapenguin.me: ELF: symbol lookup via DT_HASH](https://flapenguin.me/elf-dt-hash), this is how SysV hash section works.

Reply:
Maybe I should find how clang works to make us feel confident.

Reply:
Ok thanks, it's good for me. I'll merge the PR once the v0.11.0 will be release as it could have side effect I didn't notice yet.

Reply:
Thank you very much! Please contact me if any side effect found.